langchain/llms/llamacpp.py

```diff
@@ -100,13 +111,19 @@ def validate_environment(cls, values: Dict) -> Dict:
         use_mlock = values["use_mlock"]
         n_threads = values["n_threads"]
         n_batch = values["n_batch"]
+        use_mmap = values["use_mmap"]
         last_n_tokens_size = values["last_n_tokens_size"]
 
         try:
             from llama_cpp import Llama
 
+            if lora_path is not None:
+                use_mmap = False
```

Comment:
I'd prefer if we raise an error for incompatible parameters rather than silently altering or ignoring them here

Reply:
Do you have a link to add as a comment on why the package doesn't support memory mapping the weights for adapted models?

Reply:
@vowelparrot In [the issue that implemented loras in llama.cpp](https://github.com/ggerganov/llama.cpp/pull/820) they say "Using --lora disables mmap since the models have to be modified anyway." llama-cpp-python diverges from this behavior by allowing both, which is what causes the segfault. I [just made an issue in llama-cpp-python about this](https://github.com/abetlen/llama-cpp-python/issues/107).

I could probably remove this logic entirely and trust that llama-cpp-python will patch this and make the issue go away. I'm going to bed now, but in the morning I'll either remove these lines or add a comment and raise a ValueError instead.

Reply:
I removed these lines becaue llama-cpp-python is going to fix this issue upstream ([See here](https://github.com/abetlen/llama-cpp-python/issues/107)).